expression/util.go

```diff
@@ -560,7 +560,7 @@ func locateStringWithCollation(str, substr, coll string) int64 {
 	for {
```

Comment:
Hi, may i ask. What is the effect of this loop? Can we return below code directly?
```
index := bytes.Index(strKey, subStrKey)
	return int64(index + 1)
```

Reply:
no, some of the characters has more than one byte `weight`